### ISSUE TITLE
Node lease leak caused by lease controller swallowing error

### ISSUE THREAD

**Issue by josselin-c** [kubernetes/kubernetes#119660]
### What happened?

Hello,

We have a cluster that's showing a constant increase in node lease objects. One particularity of the leaked leases is that  they don't have an OwnerReference.

Looking at the code, the ownerReference is set by this kubelet function [`SetNodeOwnerFunc`](https://github.com/kubernetes/kubernetes/blob/dad95e1be65d4e74d33c810f4395596a22115717/pkg/kubelet/util/nodelease.go#L32) which is called by the lease controller in [`newLease`](https://github.com/kubernetes/kubernetes/blob/dad95e1be65d4e74d33c810f4395596a22115717/staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go#L230-L233). 

Note that  `newLease` returns `lease` even if the call to `newLeasePostProcessFunc` fails. The means `lease` is returned despite being potentially without an ownerRef.

Looking at `newLease()` call site in [`retryUpdateLease`](https://github.com/kubernetes/kubernetes/blob/dad95e1be65d4e74d33c810f4395596a22115717/staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go#L187) we see that the returned error is swallowed meaning the controller will make calls to `c.leaseClient.Update` with an incomplete lease.

We don't have access to the worker nodes or the kubelet logs but this explanation match what we are seeing when looking at the audit logs where the call which creates the lease without ownerRef is an Update call.

### What did you expect to happen?

Node lease should always be created with an ownerRef

### How can we reproduce it (as minimally and precisely as possible)?

To repro, setup a cluster with a worker node, delete the worker node object (but keep the node+kubelet running):
```
kubectl delete no ip-192-168-212-28.us-west-2.compute.internal
```
Then look at the lease, the original one will be removed by the GC but kubelet will eventually re-create it without an ownerRef:
```
$ kubectl -n kube-node-lease get lease ip-192-168-212-28.us-west-2.compute.internal -o json
{
    "apiVersion": "coordination.k8s.io/v1",
    "kind": "Lease",
    "metadata": {
        "creationTimestamp": "2023-07-28T21:05:05Z",
        "name": "ip-192-168-212-28.us-west-2.compute.internal",
        "namespace": "kube-node-lease",
        "resourceVersion": "1942799",
        "uid": "86429d42-7508-4929-84b2-1435ea3c6d71"
    },
    "spec": {
        "holderIdentity": "ip-192-168-212-28.us-west-2.compute.internal",
        "leaseDurationSeconds": 40,
        "renewTime": "2023-07-28T21:06:27.474137Z"
    }
}
```

### Kubernetes version

<details>

```console
$ kubectl version
Server Version: version.Info{Major:"1", Minor:"27+", GitVersion:"v1.27.3-eks-a5565ad", GitCommit:"78c8293d1c65e8a153bf3c03802ab9358c0e1a14", GitTreeState:"clean", BuildDate:"2023-06-16T17:32:40Z", GoVersion:"go1.20.5", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
Tested on EKS
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by cartermckinnon** [comment]
In this particular case, kubelet should fail to set the lease owner and return this error: https://github.com/kubernetes/kubernetes/blob/dad95e1be65d4e74d33c810f4395596a22115717/pkg/kubelet/util/nodelease.go#L49C11-L49C11

Fix looks fairly straightforward, I'll do a sanity check and put up a PR.

**Comment by odinuge** [comment]
Hi,

This seems pretty much like https://github.com/kubernetes/kubernetes/issues/109777, right?

**Comment by cartermckinnon** [comment]
Yep! Looks like a fix was attempted in #110834 but it was never merged.

**Comment by odinuge** [comment]
> Yep! Looks like a fix was attempted in https://github.com/kubernetes/kubernetes/pull/110834 but it was never merged.

Ack, lets close this then given its a duplicate of https://github.com/kubernetes/kubernetes/issues/109777.

Thanks

/close
